docs/processors/parallel-processors.mdx

```diff
@@ -0,0 +1,55 @@
+---
+title: 'Parallel Processors'
+sidebar_position: 3
+---
+
+Conduit allows you to configure a processor to process records in parallel. This
+means that multiple records can be processed simultaneously by one processor,
+which can be particularly beneficial for processors performing time-consuming
+computations on a fast stream of records.
+
+:::tip Ordering guarantee
+
+When processors are configured to process records concurrently, they guarantee
+that the order of records won't change. Although records may be processed out of
+order, the resulting order of records coming out of the processor is guaranteed
+to be unchanged.
+
+:::
+
+For instance, consider a pipeline that needs to process 100 records per second.
+One of the processors in the pipeline needs to retrieve data from a server, and
+this operation typically takes 1 second to complete. Since this operation is
+executed for every record, sequential processing won't keep up with the stream.
+However, by configuring the processor to use 200 parallel workers, we can
+execute the slow operation concurrently on 200 records at a time. This enables
+us to easily process 100 records per second and provides some room for handling
+larger bursts.
+
+## Configuration
+
+You can configure a processor to run concurrently by specifying the number of
+parallel workers. By default, the number of workers is set to 1, which means
+that records are processed sequentially.
+
+Here is an example pipeline configuration that configures the `js` processor to
+run in parallel using 10 workers:
```

Comment:
is there some kind of max number of workers?

Reply:
Yes, physics will prevent you from spawning an infinite number of workers 😄 Jokes aside, the only limit right now is the max value we can store in an `int`, on 64-bit systems that's `9223372036854775807`. I'm sure bad things would happen at far lower numbers though, so effectively there's no real limit. We should probably have one!

Reply:
yeah makes sense! was just thinking about the users and that they'd probably wonder how large of a number they can go with, so we can mention this